components/sup/src/manager/file_watcher.rs

```diff
@@ -2882,6 +2915,13 @@ mod tests {
         fn prepend_root(&self, p: &Path) -> PathBuf {
             prepend_root_impl(self.root, p, self.debug_info)
         }
+
+        fn is_poll_watcher(&self) -> bool {
+            match &self.watcher_type {
+                WatcherType::NotifyWatcherType => false,
+                WatcherType::PollWatcherType => true,
+            }
```

Comment:
do we need `watcher_type`? Could you use `watcher.get_mut_underlying_watcher().real_watcher` instead?